### PR TITLE
Add edge case tests for TextCNN model

### LINKED ISSUE #4634: Add edge case tests for TextCNN model (empty SMILES, long sequences, invalid characters)
## Summary
The TextCNN model (`deepchem/models/torch_models/text_cnn.py`) currently lacks test coverage for edge cases that users may encounter when working with real-world molecular datasets. This proposal adds three critical test cases to improve robustness and reliability.

## Current Test Coverage
The existing tests in `deepchem/models/torch_models/tests/test_textcnn_model.py` cover:
- ✅ Basic model instantiation (`test_textcnn_base`)
- ✅ Forward pass with well-formed inputs (`test_textcnn_base_forward`)
- ✅ Overfitting on small datasets (`test_textcnn_overfit_classification/regression`)
- ✅ Model serialization/reload (`test_textcnn_reload`)
- ✅ TensorFlow implementation parity (`test_textcnn_compare_with_tf_impl`)

### 1. **Empty/Very Short SMILES Strings**
**Current Behavior:** Unknown - not tested
**Expected Behavior:** Should handle gracefully or raise informative error

**Example cases:**
- Empty string: `""`
- Single atom: `"C"` 
- Minimal molecule: `"CC"`

**Why this matters:** Users might have datasets with missing or malformed entries. The model should handle these predictably.

### 2. **Very Long SMILES Sequences**
**Current Behavior:** `build_char_dict()` uses `max_length * 1.2` (line 357), but behavior beyond this is untested
**Expected Behavior:** Should handle or truncate gracefully

**Example cases:**
- SMILES > 1000 characters (large polymers, dendrimers)
- Edge case at exactly `seq_length` boundary

**Why this matters:** Large molecules (proteins, polymers) can have very long SMILES representations. Need to validate padding/truncation behavior.

### 3. **Invalid Characters in SMILES**
**Current Behavior:** `smiles_to_seq()` raises `ValueError('character not found in dict')` (line 414)
**Code reference:**
```python
else:
    raise ValueError('character not found in dict')
```

**Problem: This error path is completely untested**

**Example cases:**

Unicode characters: "C€O"
Uncommon symbols: "C@#$%"
Whitespace variations
Why this matters: Real datasets can contain corrupted or non-standard SMILES. The error handling needs validation to ensure users get clear error messages.

**Proposed Solution**
Add three new test functions to (`test_textcnn_model.py`)
```python
@pytest.mark.torch
def test_textcnn_empty_and_short_smiles():
    """Test TextCNN behavior with empty and very short SMILES strings."""
    # Test empty, single char, and minimal molecules
    
@pytest.mark.torch
def test_textcnn_long_sequences():
    """Test TextCNN with extremely long SMILES sequences."""
    # Test sequences exceeding typical lengths
    
@pytest.mark.torch
def test_textcnn_invalid_characters():
    """Test TextCNN error handling for invalid SMILES characters."""
    # Verify ValueError is raised with appropriate message
```
## Contribution
I'm interested in implementing these tests and submitting a PR. Looking forward to feedback on this approach!

### PR DESCRIPTION
**Pull request by hharshhsaini**
fix #4634

Add edge case tests for TextCNN model to improve test coverage for real-world scenarios users may encounter.

**New Tests Added:**
1. `test_textcnn_empty_and_short_smiles` - Tests empty strings, single atoms ("C"), and minimal molecules ("CC")
2. `test_textcnn_long_sequences` - Tests boundary conditions and sequences exceeding `seq_length`, validates `max_length * 1.2` calculation
3. `test_textcnn_invalid_characters` - Validates `ValueError` is raised with appropriate message for invalid SMILES characters

# Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

# Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
- [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be 0.32.0**)
- [x] Run `mypy -p deepchem` and check no errors
- [x] Run `flake8 <modified file> --count` and check no errors
- [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by hharshhsaini** [comment]
@rbharath sir please review the PR's .
